Add profile for chafa

alkim0 opened the pull request:
chafa is a terminal image viewer.

As far as I know, there is no configuration file for chafa.

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,56 @@
+notv
+nou2f
+novideo
+protocol unix
```
Note for me: Only if it has x11 access.

alkim0 replied:
It does not need x11 access. So I guess we can remove these lines and add `x11 none` after `tracelog`?

rusty-snake replied:
Yes.

- Add `x11 none`
- Add `blacklist ${RUNUSER}/wayland-?` or `blacklist ${RUNUSER}`
- Remove `protocol unix`
- Replace `seccomp` with `seccomp socket`

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,56 @@
+#include whitelist-common.inc
+include whitelist-run-common.inc
+include whitelist-runuser-common.inc
+# Users may want to view images in /usr/share
+#include whitelist-usr-share-common.inc
```
Can you check for chafa files in /usr/share and change the comment as follow.
```suggestion
# Add the folloing to your chafa.local if you do not need to view images in /usr/share.
#whitelist /usr/share/chafa  <-- or whatever is required
#include whitelist-usr-share-common.inc
```

alkim0 replied:
I'm a bit confused. Wouldn't you `whitelist /usr/share/chafa` if you *do* need to view images in /usr/share?

Regardless, chafa does not create a `/usr/share/chafa`, so I don't think this is needed.

rusty-snake replied:
> chafa does not create a /usr/share/chafa

Of course not. It would mean a normal user can write in `/usr/share`.

The (possible) files there are created by the package and read by chafa, so check either one.

rusty-snake replied:
> Wouldn't you whitelist /usr/share/chafa if you do need to view images in /usr/share?

No, it's commented. It is an opt-in for users to harden their profile if they do not need to view images in /usr/share`

alkim0 replied:
> The (possible) files there are created by the package and read by chafa, so check either one.
This is what I meant. The package (at least by Arch) does not create a `/usr/share/chafa`. It does create a `/usr/share/gtk-doc/html/chafa`, but that directory is meant to be looked at with a web browser and not an image viewer.

Should I add the line anyway? Is it preferable to:
```
#blacklist /usr/share
```

rusty-snake replied:
1. No.
2. No. It will block access to dirs like `/usr/share/terminfo` and likely break.

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,56 @@
+# Users may want to view images in ${HOME}
+#include disable-xdg.inc
+
+# Users may want to view images in ${HOME}
+#include whitelist-common.inc
```
```suggestion
```

I think we can remove it. Other image viewer profiles do not have it too.

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,56 @@
+include disable-programs.inc
+include disable-shell.inc
+include disable-write-mnt.inc
+# Users may want to view images in ${HOME}
+#include disable-xdg.inc
```
```suggestion
```

Same I guess

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,56 @@
+# Persistent global definitions
+include globals.local
+
+blacklist /usr/libexec
```
Which distro do you use? Does it use /usr/libexec?

Just to be sure it does not break.

alkim0 replied:
ArchLinux. It did not use /usr/libexec

rusty-snake reviewed (commented):
Looks good.

No firecfg?

> As far as I know, there is no configuration file for chafa.

How did you checked?

> terminal image viewer.

What is a "terminal image viewer"? Does it display the image in the terminal? So we can deny X11 access?

kmk3 commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,56 @@
+caps.drop all
+machine-id
+net none
+nodvd
```
Does it use hardware acceleration?  If not, add `no3d`.

kmk3 commented:
Project links for reference:

* Homepage: https://hpjansson.org/chafa/
* Source Code: https://github.com/hpjansson/chafa

alkim0 commented:
- Added firecfg.
- Denied x11 acess
- Added no3d

Regarding configs, the program makes no mention about a config file in the man page/project home page.
I also did a firejail --trace, and found no accesses to home.

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -19,27 +19,24 @@ include disable-write-mnt.inc
 nosound
-notv
-nou2f
-novideo
```
?

alkim0 replied:
Based on your comment above, I thought those were only needed if the program required x11 access.
Since, I set `x11 none`, I thought they were no longer needed is this not the case?

rusty-snake replied:
Why should they depend on graphical output? This are input devices.

kmk3 commented:
Could you squash/fixup the commits and force-push?

And leave only the following in the squashed commit message:

> Add profile for chafa

Example using `git rebase`:

* <https://github.com/netblue30/firejail/pull/4779#issuecomment-993932535>

Also, feel free to amend the commits and force-push, as GitHub adds a timeline
event with a link to a comparison of the commits before/after the force-push.

rusty-snake commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,54 @@
+# Persistent global definitions
+include globals.local
+
+blacklist ${RUNUSER}/wayland-*
```
```suggestion
```

alkim0 commented:
Incorporated all the latest changes (including things about x11) and readding notv, nou2f, and novideo.

kmk3 commented on etc/profile-a-l/chafa.profile:
```diff
@@ -0,0 +1,53 @@
+
+include whitelist-run-common.inc
+include whitelist-runuser-common.inc
+#include whitelist-usr-share-common.inc
```
```suggestion
# Add the following to your chafa.local if you do not need to view images in
# /usr/share.
#include whitelist-usr-share-common.inc
```

Add context to the commented code.

Based on the following comment:

* <https://github.com/netblue30/firejail/pull/5355#discussion_r961399395>

kmk3 reviewed (commented):
Nitpick: `s/Added/Add/` in the commit message.

This matches the style used on the commit messages created by git itself.
Examples:

* `git merge --no-ff` -> `Merge ...`
* `git revert` -> `Revert ...`

See the following for details:

* <https://chris.beams.io/posts/git-commit/>

Other than that, LGTM.

alkim0 commented:
Incorporated additional suggestions

kmk3 reviewed (approved):
LGTM.
